crates/metadata/src/lib.rs

```diff
@@ -57,38 +57,32 @@ use serde::{
     Serialize,
 };
 
-/// Versioned ink! project metadata.
+/// The metadata version of the generated ink! contract.
+///
+/// The serialized metadata format (which this represents) is different from the
+/// version of this crate or the contract for Rust semantic versioning purposes.
 ///
 /// # Note
 ///
-/// Represents the version of the serialized metadata *format*, which is distinct from the version
-/// of this crate for Rust semantic versioning compatibility.
-#[derive(Debug, Serialize, Deserialize)]
-#[allow(clippy::large_enum_variant)]
-pub enum MetadataVersioned {
-    /// Version 0 placeholder. Represents the original non-versioned metadata format.
-    V0(MetadataVersionDeprecated),
-    /// Version 1 of the contract metadata.
-    V1(MetadataVersionDeprecated),
-    /// Version 2 of the contract metadata.
-    V2(MetadataVersionDeprecated),
-    /// Version 3 of the contract metadata.
-    V3(InkProject),
-}
-
-impl From<InkProject> for MetadataVersioned {
-    fn from(ink_project: InkProject) -> Self {
-        MetadataVersioned::V3(ink_project)
-    }
+/// Versions other than the `Default` are considered deprecated.
+#[derive(Default, Debug, Serialize, Deserialize)]
+pub enum MetadataVersion {
+    /// Represents the original non-versioned metadata format.
+    #[serde(rename = "0")]
+    V0,
```

Comment:
I wonder whether we should remove these deprecated version variants entirely. After all they will never be constructed here, and if you want to deserialize legacy versions then you just have to use an older version of the crate.

Reply:
Yeah good point, updated